server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -123,6 +123,20 @@ public SnapshotState getSnapshotState(final SnapshotId snapshotId) {
         return indices;
     }
 
+    /**
+     * Returns the list of {@link IndexId} that have their snapshots updated but not removed (because they are still referenced by other
+     * snapshots) after removing the given snapshot from the repository.
+     *
+     * @param snapshotId SnapshotId to remove
+     * @return List of indices that are changed but not removed
+     */
+    public List<IndexId> indicesToUpdateAfterRemovingSnapshot(SnapshotId snapshotId) {
```

Comment:
Can you add a unit test for this in RepositoryDataTests?

Reply:
How about https://github.com/elastic/elasticsearch/pull/47439/commits/0961fd1062162bcb6578162bb9e36c4155555705 ?

(it's in the style of the other tests for this class, could also hard code a simpler case if you want :))

Reply:
It's good, thanks